### PR TITLE
Documentation work

### LINKED ISSUE #13: Build failing on jdk 11
I was building it on jdk 11, it seems to be failing:
```
[DEBUG]  /home/rohaan/.m2/repository/org/springframework/spring-oxm/4.2.7.RELEASE/spring-oxm-4.2.7.RELEASE.jar
[DEBUG]  /home/rohaan/.m2/repository/org/springframework/spring-test/4.2.7.RELEASE/spring-test-4.2.7.RELEASE.jar
[DEBUG]  /home/rohaan/.m2/repository/com/fasterxml/jackson/dataformat/jackson-dataformat-yaml/2.9.8/jackson-dataformat-yaml-2.9.8.jar
[DEBUG]  /home/rohaan/.m2/repository/org/yaml/snakeyaml/1.23/snakeyaml-1.23.jar
[DEBUG]  /home/rohaan/.m2/repository/junit/junit/4.11/junit-4.11.jar
[DEBUG] Source roots:
[DEBUG]  /home/rohaan/work/repos/openshift-maven-plugin/it/src/test/java
[DEBUG] Command line options:
[DEBUG] -d /home/rohaan/work/repos/openshift-maven-plugin/it/target/test-classes -classpath /home/rohaan/work/repos/openshift-maven-plugin/it/target/test-classes:/home/rohaan/work/repos/openshift-maven-plugin/it/target/classes:/home/rohaan/.m2/repository/com/consol/citrus/citrus-core/2.6.2/citrus-core-2.6.2.jar:/home/rohaan/.m2/repository/com/consol/citrus/model/citrus-model-core/2.6.2/citrus-model-core-2.6.2.jar:/home/rohaan/.m2/repository/javax/xml/bind/jaxb-api/2.2.11/jaxb-api-2.2.11.jar:/home/rohaan/.m2/repository/com/sun/xml/bind/jaxb-core/2.2.11/jaxb-core-2.2.11.jar:/home/rohaan/.m2/repository/com/sun/xml/bind/jaxb-impl/2.2.11/jaxb-impl-2.2.11.jar:/home/rohaan/.m2/repository/commons-cli/commons-cli/1.3.1/commons-cli-1.3.1.jar:/home/rohaan/.m2/repository/commons-codec/commons-codec/1.10/commons-codec-1.10.jar:/home/rohaan/.m2/repository/commons-lang/commons-lang/2.6/commons-lang-2.6.jar:/home/rohaan/.m2/repository/xalan/xalan/2.7.2/xalan-2.7.2.jar:/home/rohaan/.m2/repository/xalan/serializer/2.7.2/serializer-2.7.2.jar:/home/rohaan/.m2/repository/xerces/xercesImpl/2.11.0/xercesImpl-2.11.0.jar:/home/rohaan/.m2/repository/xml-apis/xml-apis/1.4.01/xml-apis-1.4.01.jar:/home/rohaan/.m2/repository/wsdl4j/wsdl4j/1.6.3/wsdl4j-1.6.3.jar:/home/rohaan/.m2/repository/net/sf/jtidy/jtidy/r938/jtidy-r938.jar:/home/rohaan/.m2/repository/org/hamcrest/hamcrest-core/1.3/hamcrest-core-1.3.jar:/home/rohaan/.m2/repository/org/hamcrest/hamcrest-library/1.3/hamcrest-library-1.3.jar:/home/rohaan/.m2/repository/commons-logging/commons-logging-api/1.1/commons-logging-api-1.1.jar:/home/rohaan/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.6.7/jackson-core-2.6.7.jar:/home/rohaan/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.6.7/jackson-annotations-2.6.7.jar:/home/rohaan/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.6.7/jackson-databind-2.6.7.jar:/home/rohaan/.m2/repository/net/minidev/json-smart/2.2.1/json-smart-2.2.1.jar:/home/rohaan/.m2/repository/net/minidev/accessors-smart/1.1/accessors-smart-1.1.jar:/home/rohaan/.m2/repository/org/ow2/asm/asm/5.0.3/asm-5.0.3.jar:/home/rohaan/.m2/repository/com/jayway/jsonpath/json-path/2.0.0/json-path-2.0.0.jar:/home/rohaan/.m2/repository/org/springframework/spring-core/4.2.7.RELEASE/spring-core-4.2.7.RELEASE.jar:/home/rohaan/.m2/repository/commons-logging/commons-logging/1.2/commons-logging-1.2.jar:/home/rohaan/.m2/repository/org/springframework/spring-context/4.2.7.RELEASE/spring-context-4.2.7.RELEASE.jar:/home/rohaan/.m2/repository/org/springframework/spring-aop/4.2.7.RELEASE/spring-aop-4.2.7.RELEASE.jar:/home/rohaan/.m2/repository/aopalliance/aopalliance/1.0/aopalliance-1.0.jar:/home/rohaan/.m2/repository/org/springframework/spring-beans/4.2.7.RELEASE/spring-beans-4.2.7.RELEASE.jar:/home/rohaan/.m2/repository/org/springframework/spring-expression/4.2.7.RELEASE/spring-expression-4.2.7.RELEASE.jar:/home/rohaan/.m2/repository/org/springframework/spring-jdbc/4.2.7.RELEASE/spring-jdbc-4.2.7.RELEASE.jar:/home/rohaan/.m2/repository/org/springframework/spring-tx/4.2.7.RELEASE/spring-tx-4.2.7.RELEASE.jar:/home/rohaan/.m2/repository/org/springframework/integration/spring-integration-core/4.2.9.RELEASE/spring-integration-core-4.2.9.RELEASE.jar:/home/rohaan/.m2/repository/org/springframework/retry/spring-retry/1.1.2.RELEASE/spring-retry-1.1.2.RELEASE.jar:/home/rohaan/.m2/repository/org/springframework/spring-messaging/4.2.7.RELEASE/spring-messaging-4.2.7.RELEASE.jar:/home/rohaan/.m2/repository/org/springframework/ws/spring-xml/2.2.4.RELEASE/spring-xml-2.2.4.RELEASE.jar:/home/rohaan/.m2/repository/org/springframework/spring-oxm/4.2.7.RELEASE/spring-oxm-4.2.7.RELEASE.jar:/home/rohaan/.m2/repository/org/springframework/spring-test/4.2.7.RELEASE/spring-test-4.2.7.RELEASE.jar:/home/rohaan/.m2/repository/com/fasterxml/jackson/dataformat/jackson-dataformat-yaml/2.9.8/jackson-dataformat-yaml-2.9.8.jar:/home/rohaan/.m2/repository/org/yaml/snakeyaml/1.23/snakeyaml-1.23.jar:/home/rohaan/.m2/repository/junit/junit/4.11/junit-4.11.jar: -sourcepath /home/rohaan/work/repos/openshift-maven-plugin/it/src/test/java: -g -nowarn -target 1.5 -source 1.5
[WARNING] File encoding has not been set, using platform encoding UTF-8, i.e. build is platform dependent!
[DEBUG] incrementalBuildHelper#beforeRebuildExecution
[INFO] Compiling 1 source file to /home/rohaan/work/repos/openshift-maven-plugin/it/target/test-classes
[DEBUG] incrementalBuildHelper#afterRebuildExecution
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] Source option 5 is no longer supported. Use 6 or later.
[ERROR] Target option 1.5 is no longer supported. Use 1.6 or later.
[INFO] 2 errors 
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Openshift Maven Plugin Parent 0.1-SNAPSHOT:
[INFO] 
[INFO] Openshift Maven Plugin Parent ...................... SUCCESS [  0.378 s]
[INFO] Openshift Maven Plugin ............................. SUCCESS [ 14.433 s]
[INFO] Openshift Maven :: Integration Tests ............... FAILURE [  0.541 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  15.504 s
[INFO] Finished at: 2019-06-26T17:47:22+05:30
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:testCompile (default-testCompile) on project oc-maven-it: Compilation failure: Compilation failure: 
[ERROR] Source option 5 is no longer supported. Use 6 or later.
[ERROR] Target option 1.5 is no longer supported. Use 1.6 or later.
[ERROR] -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:testCompile (default-testCompile) on project oc-maven-it: Compilation failure
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.compiler.CompilationFailureException: Compilation failure
    at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute (AbstractCompilerMojo.java:858)
    at org.apache.maven.plugin.compiler.TestCompilerMojo.execute (TestCompilerMojo.java:152)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[ERROR] 
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :oc-maven-it
~/work/repos/openshift-maven-plugin : $ javac -version
javac 11.0.2
~/work/repos/openshift-maven-plugin : $ mvn -version
Apache Maven 3.6.1 (d66c9c0b3152b2e69ee9bac180bb8fcc8e6af555; 2019-04-05T00:30:29+05:30)
Maven home: /usr/local/apache-maven/apache-maven-3.6.1
Java version: 11.0.2, vendor: Oracle Corporation, runtime: /usr/local/jdk-11.0.2
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.1.11-300.fc30.x86_64", arch: "amd64", family: "unix"
~/work/repos/openshift-maven-plugin : $ 


```

### PR DESCRIPTION
**Pull request by rohanKanojia**
also fixes #13

### CODE REVIEW

File: doc/ci-docs.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+echo ============================================
+echo Deploying openshift-maven-plugin documentation
+echo ============================================
+
+export MAVEN_OPTS="-Xmx3000m"
+
+mvn -B install -DskipTests=true
+cd doc
+mvn -B -Phtml,pdf package
+git clone -b gh-pages https://fabric8cd:$GH_TOKEN@github.com/jshiftio/openshift-maven-plugin gh-pages
```
Comment (lordofthejars) [inline doc/ci-docs.sh:28]: is this username still valid?

File: doc/src/main/asciidoc/inc/_authentication.adoc [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+  </server>
+  ....
+</servers>
+----
+
+The server id must specify the registry to push to/pull from, which by
+default is central index `docker.io` (or `index.docker.io` / `registry.hub.docker.com` as fallbacks).
+Here you should add your docker.io account for your repositories. If you have multiple accounts
+for the same registry, the second user can be specified as part of the ID. In the example above, if you
+have a second account 'fabric8io' then use an `<id>docker.io/fabric8io</id>` for this second entry. I.e. add the
```
Comment (lordofthejars) [inline doc/src/main/asciidoc/inc/_authentication.adoc]: We can change the example.

File: doc/src/main/asciidoc/inc/_global_configuration.adoc [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Global plugin configuration properties are typically apply to all Maven plugin goals.
+
+.Global plugin configuration
+[cols="1,5"]
+|===
+| Property | Description
+
+| `jshift.skip`
+| Skips over executing the goal.  Typically used to disable the Fabric8 maven plugin when it's configured in a project.
```
Comment (lordofthejars) [inline doc/src/main/asciidoc/inc/_global_configuration.adoc]: jshift maven plugin

**Review by lordofthejars — CHANGES_REQUESTED** [review]
Can you review all fabric8 mentions and check if they are still valid? For me it seems that in most cases can be renamed to jshift xxx

File: doc/ci-docs.sh (hunk 1 above)
Comment (rohanKanojia) [reply doc/ci-docs.sh:28]: Umm, we don't have any username related to jshift. Do you want me to create one? I thought maybe using existing user(http://github.com/fabric8cd) can work

File: doc/src/main/asciidoc/inc/_authentication.adoc (hunk 2 above)
Comment (rohanKanojia) [reply doc/src/main/asciidoc/inc/_authentication.adoc]: ah, yeah

**Review by lordofthejars — APPROVED** [review]

**Pull request merged**